idea/src/org/jetbrains/kotlin/idea/quickfix/DeprecatedJavaAnnotationFix.kt

```diff
@@ -40,6 +42,35 @@ internal class DeprecatedJavaAnnotationFix(element: KtAnnotationEntry,
         file.importList?.add(psiFactory.createImportDirective(ImportPath(annotationFqName, false, null)))
     }
 
+    private fun updateAnnotationArgument(file: KtFile, args: List<KtValueArgument>): String {
+        return args.joinToString(",") {
+            val type = it.text.split('.')
```

Comment:
I was trying to find a better way to do this using the PSI elements, but nothing seemed to work. I am relatively new at PSI though, so I'm very much aware that I could be missing something obvious. Please point it out if you know!

I need to handle things like `@Retention(RetentionPolicy.SOURCE)` but also `@Target(CLASS, PACKAGE, METHOD)` in a way that lets me know the type of the arguments, so that I can map the old Java arguments to their Kotlin versions.

Reply:
You could use `ktAnnotationEntry.analyze()` to get `BindingContext`, and then use `context[BindingContext.annotation, ktAnnotationEntry]` to get `AnnotationDescriptor` which contains all necessary information. Of course it's much better than parse PSI text manually. Note: to do all of this, you should pass the wholy `ktAnnotationEntry` to quick-fix constructor, not just arguments and annotation FQ name.

Reply:
Sorry for late review.

Reply:
I will look at this, sorry! I totally missed the reply here. Apologies.